src/libraries/System.Private.Xml/src/System/Xml/Serialization/TypeExtensions.cs

```diff
@@ -10,7 +11,7 @@ internal static class TypeExtensions
     {
         private const string ImplicitCastOperatorName = "op_Implicit";
 
```

Comment:
Add [NotNullWhen(true)] here?

Reply:
Actually, Carlos just pointed out the `return !targetType.IsValueType;` line, so maybe this attribute cannot be applied here.

Reply:
No, I actually initially had it this way only to find out this can actually out null in some cases